### PR TITLE
#76 Add verification endpoint

### PR DESCRIPTION
**Pull request by alieniasty**

### CODE REVIEW

**Comment by natenho** [comment]
@alieniasty thank you for your PR, it's indeed a good idea! I'll have a look at it.

**Review by natenho — COMMENTED** [review]
I've added some comments, please let me know your thoughts!

File: src/Mockaco/Startup.cs [hunk 1]
```diff
@@ -62,15 +64,24 @@ public void Configure(IApplicationBuilder app, ILogger<Startup> logger)
...
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+
+            app.UseCors("CorsPolicy");
+
+            app.MapWhen(context => !context.Request.Path.StartsWithSegments("/verify"), appBuilder =>
```
[the remark this answers is not in the document]
Comment (alieniasty) [reply src/Mockaco/Startup.cs:76]: Yes, it's possible that someone would like to mock /verify endpoint. Dynamically generated controller from a config seems like a good idea. And sounds fun, but I will need some more time to finish that ;)

File: src/Mockaco/Middlewares/RequestMatchingMiddleware.cs [hunk 2]
```diff
@@ -39,7 +41,14 @@ IEnumerable<IRequestMatcher> requestMatchers
...
             {
                 if (await requestMatchers.AllAsync(_ => _.IsMatch(httpContext.Request, mock)))
-                {                    
+                {
+                    cache.Set(mock.Route, new
+                    {
+                        Route = mock.Route,
+                        Timestamp = $"{DateTime.Now.Hour}:{DateTime.Now.Minute}",
+                        Body = await httpContext.Request.ReadBodyStream()
+                    }, DateTime.Now.AddHours(1));
```
[the remark this answers is not in the document]
Comment (alieniasty) [reply src/Mockaco/Middlewares/RequestMatchingMiddleware.cs]: Good point, I've put that into `MockacoOptions.cs`

**Comment by alieniasty** [comment]
Hey @natenho I do. I had a busy time but I am getting back to it ;)

**Comment by natenho** [comment]
@alieniasty the feature seems not to be working...could you please check?

These are the steps that I'm using to test (considering the default sample mock route : `"route": "hello/{message}"`

```
$ curl -i http://localhost:5000/hello/mock
HTTP/1.1 200 OK
Date: Sun, 14 Nov 2021 02:10:15 GMT
Content-Type: application/json
Server: Kestrel
Transfer-Encoding: chunked

{
  "id": "05168b8d-dee0-d0c8-22ee-61dec72bfd0a",
  "message": "Hello mock!",
  "createdAt": "2021-11-13T23:10:15.5504853-03:00"
}
```
```
curl -i http://localhost:5000/_mockaco/verification/?route=hello/mock
HTTP/1.1 404 Not Found
Date: Sun, 14 Nov 2021 02:11:53 GMT
Server: Kestrel
Content-Length: 0
```
I was expecting to receive a 200 OK, would you mind including some documentation/tests to the feature?

**Comment by alieniasty** [comment]
Hey @natenho I've modified the cache key to use the actual request route instead of mock route which allows to verify the mock by specific text that is put into {message} for cs script. Also, I've modified README.md with a manual.

**Comment by alieniasty** [comment]
Hey @natenho are you up for the change?

**Comment by natenho** [comment]
@alieniasty absolutely, sorry, I'll make time to review and pull it soon.

**Pull request merged**